src/group10player/Landscaper.java

```diff
@@ -78,22 +116,51 @@ public void takeTurn() throws GameActionException{
         return false;
     }*/
 
-    //Look for HQ, update foundHQ if adjacent to neighbor, and update directionHQ
+    public boolean checkWallFinished() throws GameActionException{
+        if (HQLocation == null){
+            tryFindHQLocation();
+            return false;
+        }
+        int sum = 0;
+        for (Direction dir: directions){
+            if (!rc.canSenseLocation(HQLocation.add(dir))){
+                return false;
+            }
+            sum += rc.senseElevation(HQLocation.add(dir));
+        }
+        if (sum >= (wallHeight * 8)){
+            wallFinished = true;
+            return true;
+        }
+        return false;
+    }
+
+    //Look for HQ, update directionHQ
     public boolean findHQ() throws GameActionException{
         if (HQLocation == null){
-            walkRandom();
             tryFindHQLocation();
+            walkRandom();
             return false;
         }
+        int minElevation = 1000;
```

Comment:
What is the minElevation 1000 for?

Reply:
That's just for comparing with other elevations - I just picked a number that would be higher than any elevation we'd ever find in the map.